src/CoreWCF.RabbitMQ.Client/src/CoreWCF/ServiceModel/Channels/RabbitMqChannelFactory.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System.Linq;
+using System;
+using System.ServiceModel.Channels;
+using System.Threading.Tasks;
+
+namespace CoreWCF.ServiceModel.Channels
+{
+    public class RabbitMqChannelFactory : ChannelFactoryBase<IOutputChannel>
+    {
+        private BufferManager _bufferManager;
+        private MessageEncoderFactory _messageEncoderFactory;
+        private RabbitMqTransportBindingElement _transport;
+
+        internal RabbitMqChannelFactory(RabbitMqTransportBindingElement transport, BindingContext context)
+            : base(context.Binding)
+        {
+            _transport = transport;
+            _bufferManager = BufferManager.CreateBufferManager(transport.MaxBufferPoolSize, int.MaxValue);
+
+            var messageEncoderBindingElements = context.BindingParameters.OfType<MessageEncodingBindingElement>().ToList();
+            if (messageEncoderBindingElements.Count > 1)
```

Comment:
Did you copy this code from somewhere as I want to check if it needs to be fixed where you copied it from? The more efficient way to do this is to call `context.BindingParameters.Find<MessageEncodingBindingElement>()` and do a null check. You could also use FindAll which returns a collection and do the count check.

Reply:
I no longer have the source, but it was from another WCF Client, outside of dotnet/wcf. BindingParameters here is of type `System.ServiceModel.Channels.BindingParameterCollection` which does not seem to have the `Find()` or `FindAll()` methods.

Reply:
[BindingParameterCollection](https://github.com/dotnet/wcf/blob/main/src/System.ServiceModel.Primitives/ref/System.ServiceModel.Primitives.cs#L1013) derives from KeyedByTypeCollection\<object\> which has a [Find\<T\>() method](https://github.com/dotnet/wcf/blob/main/src/System.ServiceModel.Primitives/ref/System.ServiceModel.Primitives.cs#L54).

Reply:
I might be missing something, but it looks like the change to have `BindingParameterCollection` derive from `KeyedByTypeCollection` wasn't released yet. We're referencing the latest System.ServiceModel.Primitives pkg (v4.10, released Aug 17, 2022), and BindingParameterCollection still derives from `KeyedCollection`. Looks closer to [this version](https://github.com/dotnet/wcf/blob/3f393620ec22b553dff47f999068d64db8ab39f1/src/System.ServiceModel.Primitives/ref/System.ServiceModel.Primitives.cs#L992).